GamebookEngine/Views/Editing/Page List/PagesTableViewController.swift

```diff
@@ -72,8 +88,40 @@ class PagesTableViewController: UITableViewController {
     }
 
     override func tableView(_: UITableView, didSelectRowAt indexPath: IndexPath) {
-        guard let page = pages.item(at: indexPath.row) else { return }
-        delegate?.selectedPage(page)
+        if searching {
+            guard let filteredPage = filteredPages.item(at: indexPath.row) else { return }
+            delegate?.selectedPage(filteredPage)
+        } else {
+            guard let page = pages.item(at: indexPath.row) else { return }
+            delegate?.selectedPage(page)
+        }
+
         navigationController?.popViewController(animated: true)
     }
+
+    // MARK: Search Bar
+
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        // Leaves function if the entered text contains whitespace
+        if searchText.trimmingCharacters(in: .whitespaces).isEmpty {
+            searching = false
+            tableView.reloadData()
+            return
+        }
+
+        filteredPages = [Page]()
+
+        for page in pages {
+            if page.content.lowercased().contains(searchText.lowercased()) {
+                filteredPages.append(page)
+            }
+        }
```

Comment:
> While this works, I am wondering if this is going to be a performant option when dealing with very large gamebooks? We're unlikely to run into gamebooks that weigh in at several megabytes, and even then maybe it won't matter.

We could definitely test it on a larger gamebook.

> There's also the consideration of reuse. What if when we add search to the game overview screen, we can't easily reuse this search functionality in another place, since it's baked into the view? There's an argument to be made as well that the views shouldn't contain any real business logic, so even if we don't change how this works, it should be refactored into GameDatabase, which'll make it reusable elsewhere.

I agree. Reusability is always optimal. Creating a method inside GameDatabase would work best and can be called from the PagesTableViewController.

Reply:
There was a point where I was considering, for testing purposes, creating a "random gamebook generator" that could create a gamebook with **a lot** of pages and routes in it. That could be a fun project for you maybe? I'm not sure how we'd integrate it into the app, maybe have a special screen somewhere for debug options.

Reply:
> I'm not sure how we'd integrate it into the app, maybe have a special screen somewhere for debug options.

If we are using it for testing purposes, we could include the generated gamebooks into test suites if that is possible. At least, that is the way I can think of unless there is something else we need to do that falls outside the scope of Xcode's testing utilities.

Reply:
Yeah I haven't done anything with Xcode unit tests before. I have the barebones template files set up in this project but I haven't used them. If you want to write some tests and see how it works, go for it! I'd also like to see the generated gamebooks available for manual testing too, though, just so we can see how very large books feel in the real world.

Reply:
Feel free to create an issue (and/or card in the Beta project) for "Debug Screen / Generating Gamebooks". Always good to catalog what you plan to work on in the future just for reference, if that's a project you wanna take on.

Reply:
Refactor looks good! Going to resolve this one, too.